game-core/src/main/java/games/strategy/engine/data/properties/MapProperty.java

```diff
@@ -29,26 +29,28 @@ public MapProperty(final String name, final String description, final Map<T, U>
   }
 
   @SuppressWarnings("unchecked")
-  private void resetProperties(final Map<T, U> map, final List<IEditableProperty> properties, final String name,
+  private void resetProperties(final Map<T, U> map, final List<IEditableProperty<U>> properties, final String name,
       final String description) {
     properties.clear();
     for (final Entry<T, U> entry : map.entrySet()) {
       final String key = (String) entry.getKey();
       final U value = entry.getValue();
       if (value instanceof Boolean) {
-        properties.add(new BooleanProperty(key, description, ((Boolean) value)));
+        properties.add((IEditableProperty<U>) new BooleanProperty(key, description, ((Boolean) value)));
```

Comment:
I'm still inclined to think it's wrong for this class to store values typed as anything other than `Object` simply because this method demonstrates `properties` is intended to be a heterogeneous map.

Also, line 36 above seems to imply that `T` is always `String`, but that might not be true given the code in `validate()`.

So, I'm still thinking this class should actually be holding onto a `Map<Object, Object>` or a `Map<String, Object>` if the latter is possible.

Reply:
Yeah, that's fine.  The generic map was already there, so the problem pre-existed this PR.  This change just made it more noticeable. :smile:

Reply:
I kind of recant my assertion that the value type of the map should be `Object`.  I started to write some unit tests for the `MapProperty#validate()` method and discovered that the implementation forces the value type of all entries to be the same (at least according to the LSP).  So, theoretically, the map values would be homogeneous.

Interesting to note, however, that the validation check doesn't occur via the constructor, so one could slip in a heterogeneous map unintentionally.